src/Hack.sol

```diff
@@ -626,55 +629,74 @@ contract WinnerTakesAll {
         _;
     }
 
-    /// @dev Creates new rounds.
-    /// @param _numberOfRounds The number of rounds to create.
-    function createNewRounds(uint256 _numberOfRounds) external {
-        for (uint256 i = 0; i < _numberOfRounds; i++) {
-            rounds.push();
+    /// @dev Creates new lotteries.
+    /// @param numberOfLotteries The number of lotteries to create.
+    function createNewLotteries(uint256 numberOfLotteries) external {
+        for (uint256 i = 0; i < numberOfLotteries; i++) {
+            lotteries.push();
         }
     }
 
+    /// @dev Buys a ticket for a participant.
+    /// @param lotteryIndex The index of the round concerned.
+    function buyTicketForLottery(uint256 lotteryIndex) external payable {
+        require(msg.value == 1 ether, "wrong value");
+        lotteries[lotteryIndex].ticketNumber++;
+        lotteries[lotteryIndex].ticketDistribution[msg.sender].push(lotteries[lotteryIndex].ticketNumber);
+    }
+
     /// @dev Set the reward at a specific round.
-    /// @param _roundIndex The index of the round concerned by the reward.
-    function setRewardsAtRound(uint256 _roundIndex) external payable onlyOwner {
-        require(rounds[_roundIndex].rewards == 0);
-        rounds[_roundIndex].rewards = msg.value;
-    }
-
-    /// @dev Allows the participation of a set of addresses for a specific round.
-    /// @param _roundIndex The index of the round concerned.
-    /// @param _recipients The set of addresses allowed to participate.
-    function setRewardsAtRoundFor(uint256 _roundIndex, address[] calldata _recipients) external onlyOwner {
-        for (uint256 i; i < _recipients.length; i++) {
-            rounds[_roundIndex].isAllowed[_recipients[i]] = true;
-        }
+    /// @param lotteryIndex The index of the round concerned by the reward.
+    function setRewardsAtRound(uint256 lotteryIndex) external payable onlyOwner {
+        require(lotteries[lotteryIndex].rewards == 0);
+        lotteries[lotteryIndex].rewards = msg.value;
     }
 
-    /// @dev Checks if an address can participate in this round.
-    /// @param _roundIndex The index of the round to be checked.
-    /// @param _recipient The address whose authorization is to be checked.
-    function isAllowedAt(uint256 _roundIndex, address _recipient) external view returns (bool) {
-        return rounds[_roundIndex].isAllowed[_recipient];
+    /// @dev Set the winning number. It is chosen randomly off-chain by the trusted owner.
+    /// @param lotteryIndex The index of the round concerned.
+    /// @param winningNumber The winning number of the lottery.
+    function setWinningNumberAtRound(uint256 lotteryIndex, uint256 winningNumber) external onlyOwner {
+        require(winningNumber <= lotteries[lotteryIndex].ticketNumber, "Incorrect winning ticket");
+        require(winningNumber != 0, "Incorrect winning ticket");
```

Comment:
why it can't be 0?

Reply:
because the 0 is not given to the participant. And it is already initialized at 0.